docs/understanding-preprocessing-codes.md

```diff
@@ -0,0 +1,490 @@
+
+seq_growth_first <- seq(from= 0, by=as.numeric(unname(growth_year)),length.out = age_stop_growth+1)
+seq_growth_last <- rep(seq_growth_first[length(seq_growth_first)],age_max_curves - age_stop_growth)
+seq_growth <- c(seq_growth_first,seq_growth_last)
+
+growth_curve <- c("Promedio Regional","Promedio Regional",Origen,AIDBSPP,seq_growth)
+growth <- rbind(growth,growth_curve)
+```
+
+We can then create the growth curves for the Bosque Inicial (Initial Forest). Right now we are using assign the Red Alder species, which will be replaced afterwards with Chilean generic species parameters. We go for all combinations of forest type and structure and the growth curve will give a fixed value of volume for all the forests in the area:
+
+```R
+Origen <- "Bosque Inicial"
+AIDBSPP <- "Red alder"
+
+for (Tipofor in forest_types){
+  for (Estruc in structures){
+    growth_year <- regional_vol / age_stop_growth
+    seq_growth_first <- seq(from= 0, by=as.numeric(unname(growth_year)),length.out = age_stop_growth+1)
+    seq_growth_last <- rep(seq_growth_first[length(seq_growth_first)],age_max_curves - age_stop_growth)
+    seq_growth <- c(seq_growth_first,seq_growth_last)
+    growth_curve <- c(Tipofor,Estruc,Origen,AIDBSPP,seq_growth)
+    growth <- rbind(growth,growth_curve)
+  }
+}
+```
+
+In a similar manner, we can add extra classifiers curves for the initial forest. For non-stocked (non-forest) pixels, we can use the "non stocked" species so the GCBM will ignore the growth:
+
+```R
+seq_growth <- rep(0,age_max_curves + 1 )
+growth_curve <- c("No forestal","No forestal","No forestal","Not stocked",seq_growth)
+growth <- rbind(growth,growth_curve)
+```
+
+The names are finally assigned to the dataframe and the volume values are converted to numeric and appended to the dataframe. The CSV files are written and the growth curve is created.
+
+### Inventory Layer
+
+The Inventory Layer file creates the initial inventory layer. The `input_traza`, `input_SOC` and the `output_gcbm` parameters define the Trazabilidad file where the land usage data is stored, the folder where the soil organic carbon data is stored and the folder of the output data respectively.
+
+Apart from this, we define the `inventory_year` which denotes the initial year of model execution. The `layer_traza` and `layer_SOC` defines the land use data (Trazabilidad) and the Soil organic carbon data from FAO respectively.
+
+We will be generating the inventory layer from the trazabilidad file by reading the same and optionally filtering the database to accommodate the polygons that were forest at a particular point in the time-series data. Further, the classifiers are assigned, which includes the Tipofor classifier, Forest structure classifier and Origin classifier.
+
+The Tipofor classifier gets the forest types from the 1997 map and assigns the "Bosque Mixto" and "Matorral Arborescente" forest type from the T1 column. All polygons without forest types will be left as "No Forestal".
```

Comment:
Polygons that are "non forest" will be left as No Forestal

Reply:
Done!